File: 0609 Find Duplicate File in System/0609.cpp
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        unordered_map<string, vector<string>> mp;

        for (string path: paths){
            istringstream ss(path);
            string folder;
            ss >> folder;

            while (ss){
                string s;
                ss >> s;
                if (s.size() > 0){
                    int i = s.find('(');
                    string filename = s.substr(0, i); 
                    string content = s.substr(i); 
                    mp[content].push_back(folder + "/" + filename);
                }
            }
        }
        
        vector<vector<string>> ans;
        
        for (auto x: mp){
            if (x.second.size() > 1){
                ans.push_back(x.second);
            }
        }
        
        return ans;
    }
};

File: 0609 Find Duplicate File in System/0609.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        d = defaultdict(list)

        for path in paths:
            path = path.split()
            folder = path[0]
            for s in path[1:]:
                filename, content = s.split('(')
                content = content[:-1]
                d[content].append(f"{folder}/{filename}")
        
        ans = []

        for content in d:
            if len(d[content]) > 1:
                ans.append(d[content])

        return ans
